pkg/pgmodel/cache/inverted_labels_cache.go

```diff
@@ -0,0 +1,63 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/ristretto"
+)
+
+type LabelInfo struct {
+	LabelID int32
+	Pos     int32
+}
+
+type LabelKey struct {
+	MetricName, Name, Value string
+}
+
+func NewLabelKey(metricName, name, value string) LabelKey {
+	return LabelKey{MetricName: metricName, Name: name, Value: value}
+}
+
+// Cache hash function needs string to work with
+func (lk LabelKey) String() string {
+	return fmt.Sprintf("%s#%s=%s", lk.MetricName, lk.Name, lk.Value)
+}
+
+func NewLabelInfo(lableID, pos int32) LabelInfo {
+	return LabelInfo{LabelID: lableID, Pos: pos}
+}
+
+// Label key-pair -> (id,pos) cache
+// Used when creating series to avoid DB calls for labels
+type InvertedLabelsCache struct {
+	cache *ristretto.Cache
```

Comment:
Why a new cache library? Why not use the clock cache?

Reply:
I agree here, we should either stick to the one we use everywhere or evaluate if we need a different cache (unlikely).

Reply:
I was actually thinking to replace our own caching impl with some of more popular caching libs. I've did some benchmarking and found that performance are pretty much the same so yeah we can keep using our own caching...